src/d3a/models/strategy/update_frequency.py

```diff
@@ -203,13 +202,19 @@ def _calculate_price_decrease_rate(self, market):
             reduced_price = \
                 self.calculate_initial_sell_rate(market.time_slot) - \
                 price_dec_per_update * self._price_update_interval
+            if reduced_price < self.final_selling_rate:
```

Comment:
Weird that this did not come up earlier...

Reply:
That is indeed something I did not figure out too, why it remained undetected.